### ISSUE TITLE
Turkish locale fail on weekdaysParse

### ISSUE THREAD

**Issue by wadouk** [moment/moment#2920]
https://jsfiddle.net/xt4d0rLh/4/

Sunday and monday as friday and saturday starts as the same in turkish locale.

What I want is to build an array where the first element depends on the first day of week of the locale and the tail of the array is the other days in the right order.

**Comment by wadouk** [comment]
That could be done by that 
``` javascript

  const weekdays = moment.weekdays();
  const firstDayOfWeek = moment.localeData().firstDayOfWeek();
  const firstDayOfWeekToEnd = weekdays.slice(firstDayOfWeek);
  const beforeFirstDayOfWeek = weekdays.slice(0, firstDayOfWeek);
  return firstDayOfWeekToEnd.concat(beforeFirstDayOfWeek);
```

**Comment by Oire** [comment]
@mj1856, @BYK, @erhangundogan, any thoughts? The locale seems to be correct, the test do not fail.

**Comment by BYK** [comment]
@Oire looks like `parse` tests only cover month names.

**Comment by Oire** [comment]
@BYK,   so   could  you  please  update  the tests and see what fails?
Teşekkür ederim!

**Comment by BYK** [comment]
@Oire - submitted a PR with a test failing on `develop` and passing with the updates. Had to disable a line in the new test since the `i` flag in `RegEx` is not locale aware so it doesn't match `SALI` when it is passed `Salı` (note the lack of dot in the lower-case version).

**Comment by Oire** [comment]
@BYK,  oh, I do know the Turkish İ/i/I/ı localization pain :(. I think, this fix is
satisfactory for you as a native speaker. Thanks again!
P.S. Türkçe küçük biliyorum :)

**Comment by Oire** [comment]
Closed via #2988.

**Comment by BYK** [comment]
@Oire - Haha thanks! The fix is okay. Also the `İ/i/I/ı` was already there, I'm just fixing what @wadouk reported. FYI this changes the behavior of non-strict parsing a bit as you may have realized. No existing test failed though, so assuming it's okay.
